components/automate-deployment/pkg/backup/builder_operations.go

```diff
@@ -0,0 +1,253 @@
+package backup
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type BuilderMinioDumpOperation struct {
+	Name       string   `json:"name"`
+	ObjectName []string `json:"storage_key"`
+}
+
+var _ Operation = &BuilderMinioDumpOperation{}
+
+func (d *BuilderMinioDumpOperation) Backup(backupCtx Context, om ObjectManifest, progChan chan OperationProgress) error {
+	backupBucketPrefix := d.backupBucketPrefix()
+	ctx := backupCtx.ctx
+
+	logrus.WithFields(logrus.Fields{
+		"name":      d.Name,
+		"backup_id": backupCtx.backupTask.TaskID(),
+		"prefix":    backupBucketPrefix,
+		"operation": "builder_minio_dump",
+		"action":    "backup",
+	}).Info("Running backup operation")
+
+	objects, _, err := backupCtx.builderBucket.List(ctx, "", false)
+	if err != nil {
+		return err
+	}
+	builderArtifacts, err := ioutil.TempFile("", "builderartifacts")
+	if err != nil {
+		return err
+	}
+	defer builderArtifacts.Close()
+	defer os.Remove(builderArtifacts.Name())
+
+	objVerifier := &NoOpObjectVerifier{}
+	for _, obj := range objects {
+		if err := d.copyObjectFromBuilder(backupCtx, backupBucketPrefix, obj, objVerifier); err != nil {
+			return err
+		}
+		if _, err := fmt.Fprintln(builderArtifacts, obj.Name); err != nil {
+			return err
+		}
+	}
+	// Write file with artifact list
```

Comment:
Instead of this comments, I suppose we could extract this to a function.  I'm fine with it as is, but if we don't extract it.  Perhaps:

```suggestion
	// Write file with artifact list
	//
	// We first seek to the beginning of the file so io.Copy works as expected.
```